code/modules/mod/modules/modules_general.dm

```diff
@@ -636,3 +636,99 @@
 
 /obj/item/mod/module/plasma_stabilizer/on_unequip()
 	REMOVE_TRAIT(mod.wearer, TRAIT_NOSELFIGNITION, MOD_TRAIT)
+
+//Finally, https://pipe.miroware.io/5b52ba1d94357d5d623f74aa/mspfa/Nuke%20Ops/Panels/0648.gif can be real:
+///Hat Stabilizer - Allows displaying a hat over the MOD-helmet, à la plasmamen helmets.
+/obj/item/mod/module/hat_stabilizer
+	name = "MOD hat stabilizer module"
+	desc = "A simple set of deployable stands, directly atop one's head; \
+	these will deploy under a select few hats to keep them from falling off, allowing them to be worn atop the sealed helmet.\
+	\n	You still need to take the hat off your head while the helmet deploys, though.\
+	\n	This is a must-have for Nanotrasen Captains, enabling them to show off their authoritative hat even while in their MODsuit."
+	icon_state = "hat_holder"
+	incompatible_modules = list(/obj/item/mod/module/hat_stabilizer)
+	/*Intentionally left inheriting 0 complexity and removable = TRUE;
+	even though it comes inbuilt into the Magnate/Corporate MODS and spawns in maints, I like the idea of stealing them*/
+	///Currently "stored" hat. No armor or function will be inherited, ONLY the icon.
+	var/obj/item/clothing/head/attached_hat
+	///Whitelist of attachable hats; read note in Initialize() below this line
+	var/static/list/attachable_hats_list
+
+/obj/item/mod/module/hat_stabilizer/Initialize()
+	. = ..()
+	attachable_hats_list = typecacheof(
+	//List of attachable hats. Make sure these and their subtypes are all tested, so they dont appear janky.
+	//This list should also be gimmicky, so captains can have fun. I.E. the Santahat, Pirate hat, Tophat, Chefhat...
+	//Yes, I said it, the captian should have fun.
+		list(
+			/obj/item/clothing/head/caphat,
+			/obj/item/clothing/head/crown,
+			/obj/item/clothing/head/centhat,
+			/obj/item/clothing/head/centcom_cap,
+			/obj/item/clothing/head/pirate,
+			/obj/item/clothing/head/santa,
+			/obj/item/clothing/head/hardhat/reindeer,
+			/obj/item/clothing/head/sombrero,
+			/obj/item/clothing/head/kitty,
+			/obj/item/clothing/head/rabbitears,
+			/obj/item/clothing/head/festive,
+			/obj/item/clothing/head/powdered_wig,
+			/obj/item/clothing/head/weddingveil,
+			/obj/item/clothing/head/that,
+			/obj/item/clothing/head/nursehat,
+			/obj/item/clothing/head/chefhat,
+			/obj/item/clothing/head/papersack,
+			)) - /obj/item/clothing/head/caphat/beret
```

Comment:
subtract the beret before you typecache?

Reply:
I've no clue how I'd go about doing that

Reply:
nevermind